src/dependency-registry.ts

```diff
@@ -0,0 +1,388 @@
+  dependency.preferredRepository = event.params._preferredRepository;
+  dependency.additionalRepositoryCount = BigInt.fromI32(0);
+  dependency.referenceWebsite = event.params._referenceWebsite;
+  dependency.scriptCount = BigInt.fromI32(0);
+}
+
+export function handleDependencyRemoved(event: DependencyRemoved): void {
+  const dependency = Dependency.load(event.params._dependencyType.toString());
+  if (dependency) {
+    store.remove("Dependency", dependency.id);
+  }
+}
+
+export function handleDependencyPreferredCDNUpdated(
+  event: DependencyPreferredCDNUpdated
+): void {
+  const dependency = Dependency.load(event.params._dependencyType.toString());
+  if (dependency) {
+    dependency.preferredCDN = event.params._preferredCDN;
+    dependency.save();
+  }
+}
+
+export function handleDependencyPreferredRepositoryUpdated(
+  event: DependencyPreferredRepositoryUpdated
+): void {
+  const dependency = Dependency.load(event.params._dependencyType.toString());
+  if (dependency) {
+    dependency.preferredRepository = event.params._preferredRepository;
+    dependency.save();
+  }
+}
+
+export function handleDependencyReferenceWebsiteUpdated(
+  event: DependencyReferenceWebsiteUpdated
+): void {
+  const dependency = Dependency.load(event.params._dependencyType.toString());
+  if (dependency) {
+    dependency.referenceWebsite = event.params._referenceWebsite;
+    dependency.save();
+  }
+}
+
+export function handleDependencyAdditionalCDNUpdated(
+  event: DependencyAdditionalCDNUpdated
+): void {
+  const dependency = Dependency.load(event.params._dependencyType.toString());
+  if (dependency) {
+    const id = generateDependencyAdditionalCDNId(
+      dependency.id,
+      event.params._additionalCDNIndex
+    );
+
+    let additionalCDN = DependencyAdditionalCDN.load(id);
+
+    if (!additionalCDN) {
+      additionalCDN = new DependencyAdditionalCDN(id);
+      dependency.additionalCDNCount = dependency.additionalCDNCount.plus(
+        BigInt.fromI32(1)
+      );
+      dependency.save();
+    }
+
+    additionalCDN.dependency = dependency.id;
+    additionalCDN.cdn = event.params._additionalCDN;
+    additionalCDN.index = event.params._additionalCDNIndex;
+    additionalCDN.save();
+  }
+}
+
+export function handleDependencyAdditionalCDNRemoved(
+  event: DependencyAdditionalCDNRemoved
+): void {
+  const dependency = Dependency.load(event.params._dependencyType.toString());
+  if (dependency) {
+    const id = generateDependencyAdditionalCDNId(
+      dependency.id,
+      event.params._additionalCDNIndex
+    );
+
+    const additionalCDN = DependencyAdditionalCDN.load(id);
+    if (!additionalCDN) {
+      return;
+    }
+    store.remove("DependencyAdditionalCDN", additionalCDN.id);
+
+    // Load each additional CDN after the removed one and decrement their index
+    for (
+      let i = event.params._additionalCDNIndex.toI32() + 1;
+      i < dependency.additionalCDNCount.toI32();
+      i++
+    ) {
```

Comment:
Based on the smart contract's behavior, I think this has to be altered to move the dependency at the last index to `event.params._additionalCDNIndex`, instead of sliding all dependencies higher than the removed dependency down by 1.

Reply:
Oops wasn't thinking here! Good catch!